File: from-3401-to-3500/3467-transform-array-by-parity/solution.py
class Solution:
    def transformArray(self, nums: List[int]) -> List[int]:
        n = len(nums)
        result = [0] * n
        ptr = n - 1
        for num in nums:
            if num % 2:
                result[ptr] = 1
                ptr -= 1
        return result

File: from-3401-to-3500/3467-transform-array-by-parity/solution.go
func transformArray(nums []int) []int {
    n := len(nums)
    result := make([]int, n)
    ptr := n - 1
    for i := 0; i < n; i ++ {
        if nums[i] % 2 == 1 {
            result[ptr] = 1
            ptr -= 1
        }
    }
    return result
}